### PR TITLE
boards: added support for the Nordic nRF51 Dongle

### PR DESCRIPTION
**Pull request by haukepetersen**
rebased on #3207

As @PeterKietzmann was so nice to send me one of this dongles, I could just not go without porting it...

NOTE: For flashing I had to update my version of the [Segger `JLinkExe` tool](https://www.segger.com/jlink-software.html), I am now using version 
```
SEGGER J-Link Commander V5.00c ('?' for help)
Compiled Jun 11 2015 11:42:46
DLL version V5.00c, compiled Jun 11 2015 11:42:43
```
My old version (4.8x something) did not work with this board...

### CODE REVIEW

**Comment by haukepetersen** [comment]
rebased

File: boards/nrf51dongle/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# define the used CPU
+export CPU = nrf51
+export CPU_MODEL = nrf51x22xxab
```
Comment (PeterKietzmann) [inline boards/nrf51dongle/Makefile.include]: Where did you find the cpu-model of dongle I sent you?

**Comment by PeterKietzmann** [comment]
I did not look into details but the `hwtimer_wait` test seems to take 7-8 seconds instead of 4 (as described by the application). 
```
2015-07-20 13:22:15,327 - INFO # kernel_init(): jumping into first task...
2015-07-20 13:22:15,332 - INFO # This is a regression test for a race condition in hwtimer_wait.
2015-07-20 13:22:15,339 - INFO # When the race condition is hit, the timer will wait for a very very long time.
2015-07-20 13:22:15,342 - INFO # The test should take about 4 sec.
2015-07-20 13:22:22,863 - INFO # success
```

**Comment by PeterKietzmann** [comment]
All in all the PR looks good and worked well! Once Travis is happy I'd like to merge it.

**Comment by PeterKietzmann** [comment]
(Therefore you need to rebase because of #3454)

**Comment by haukepetersen** [comment]
rebased.

File: boards/nrf51dongle/Makefile.include (hunk 1 above)
Comment (haukepetersen) [inline boards/nrf51dongle/Makefile.include]: magnifying lenses... :-)

**Comment by PeterKietzmann** [comment]
@haukepetersen some build tests are failing. One in the x86 group which is unrelated and several in the cortex m0 group. I think you need to add a `BOARD_INSUFFICIENT_RAM` entry for the regarding applications

**Comment by haukepetersen** [comment]
blacklisted board for failing tests as it has insufficient RAM. Hope I didn't miss any.

**Comment by PeterKietzmann** [comment]
Travis will tell us...

**Comment by haukepetersen** [comment]
thumbs pressed :-)

**Comment by haukepetersen** [comment]
Travis is happy.

**Comment by PeterKietzmann** [comment]
How should we deal with strider? It does not finish, even if it seems to have passed successfully.

**Comment by haukepetersen** [comment]
Strider is optional, Travis is the merging criteria..

**Comment by PeterKietzmann** [comment]
Well, fine. Then I'll skip this optional step, as its output looks good. ACK and go

**Pull request merged**